### PR TITLE
Restrict StateTransitionerFactory

### PR DESCRIPTION
**Pull request by axchu**
### Fixes
- Fixes https://github.com/ethereum-optimism/roadmap/issues/202

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/verification/OVM_StateTransitionerFactory.sol [hunk 1]
```diff
@@ -1,17 +1,27 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.7.0;
 
+/* Logging */
+import { console } from "@nomiclabs/buidler/console.sol";
```
Comment (karlfloersch) [inline contracts/optimistic-ethereum/OVM/verification/OVM_StateTransitionerFactory.sol:5]: Forgot to delete `console.log`

File: test/contracts/OVM/verification/OVM_StateTransitionerFactory.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  let OVM_StateTransitionerFactory: Contract
+  beforeEach(async () => {
+    OVM_StateTransitionerFactory = await Factory__OVM_StateTransitionerFactory.deploy(
+      AddressManager.address
+    )
+  })
+
+  describe('create', () => {
+    describe("when the sender is not the OVM_FraudVerifier", () => {
```
Comment (tynes) [inline test/contracts/OVM/verification/OVM_StateTransitionerFactory.spec.ts:39]: nit: mixing `"` and `'` for strings

**Comment by kfichter** [comment]
Hey @axchu! Integration tests are failing because this PR changes the constructor for `OVM_StateTransitionerFactory`, which means we'll also have to make changes to the deployment config.

Specifically, the following line:

https://github.com/ethereum-optimism/contracts-v2/blob/6bc0c48222579d9e49c70a5ad270c818c58cf24e/src/contract-deployment/config.ts#L143-L145

Should include the address manager as a parameter. You can see an example of how this is done here:

https://github.com/ethereum-optimism/contracts-v2/blob/6bc0c48222579d9e49c70a5ad270c818c58cf24e/src/contract-deployment/config.ts#L146-L149

Everything should be passing once this is done!

**Comment by kfichter** [comment]
OK lgtm now!

**Pull request merged**